firmware/lib/screenManager/screenManager.cpp

```diff
@@ -26,12 +32,54 @@ ScreenManager::ScreenManager(TFT_eSPI &tft) : m_tft(tft) {
   Serial.println("SCREEN_3_CS:" + String(SCREEN_3_CS));
   Serial.println("SCREEN_4_CS:" + String(SCREEN_4_CS));
   Serial.println("SCREEN_5_CS:" + String(SCREEN_5_CS));
+
+}
+
+void ScreenManager::setFont(TTF_Font font) {
+  if (font == m_curFont) {
+    // nothing to do
+    return;
+  }
+  m_render.unloadFont();
+  if (font == TTF_Font::NONE) {
+    // just unload
+    return;
+  }
+  // 0 is success
+  FT_Error error = 1;
+  switch (font)
+  {
+  case ROBOTO_REGULAR:
+    error = m_render.loadFont(robotoRegular_start, robotoRegular_end-robotoRegular_start);
+    break;
+  
+  case FINAL_FRONTIER:
+    error = m_render.loadFont(finalFrontier_start, finalFrontier_end-finalFrontier_start);
+    break;
+
+  case DSEG7:
+    error = m_render.loadFont(dseg7_start, dseg7_end-dseg7_start);
+    break;
+
+  case DSEG14:
+    error = m_render.loadFont(dseg14_start, dseg14_end-dseg14_start);
+    break;
+  }
+  if (error == 0) {
+    m_curFont = font;
+  } else {
+    Serial.printf("Unable to load TTF font %d\n", font);
+  } 
 }
 
 TFT_eSPI &ScreenManager::getDisplay() {
   return m_tft;
 }
 
+OpenFontRender &ScreenManager::getRender() {
```

Comment:
Does this renderer need to be exposed?

Reply:
I would like to set .getRender() and .getDisplay() to private.
Then all drawing would have to go through ScreenManager.

It did not do it, because I've not implemented delegates for all TFT_eSPI and OpenFontRender functions yet.
But we could set them to private and just add the rest of the delegate functions when we need them.